litestar/_openapi/plugin.py

```diff
@@ -1,22 +1,55 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any, Dict
 
 from litestar._openapi.datastructures import OpenAPIContext
 from litestar._openapi.path_item import create_path_item_for_route
-from litestar.exceptions import ImproperlyConfiguredException
+from litestar.di import Provide
+from litestar.enums import MediaType
+from litestar.exceptions import ImproperlyConfiguredException, NotFoundException
+from litestar.handlers import get
 from litestar.plugins import InitPluginProtocol
 from litestar.plugins.base import ReceiveRoutePlugin
+from litestar.response import Response
+from litestar.router import Router
 from litestar.routes import HTTPRoute
+from litestar.status_codes import HTTP_404_NOT_FOUND
 
 if TYPE_CHECKING:
     from litestar.app import Litestar
     from litestar.config.app import AppConfig
+    from litestar.connection import Request
+    from litestar.handlers import HTTPRouteHandler
     from litestar.openapi.config import OpenAPIConfig
+    from litestar.openapi.plugins import OpenAPIRenderPlugin
     from litestar.openapi.spec import OpenAPI
     from litestar.routes import BaseRoute
 
 
+def handle_not_found(path: str = "/") -> Response:
+    """Handler for returning HTML formatted errors from not-found schema paths.
+
+    This preserves backward compatibility with the Controller-based OpenAPI implementation.
+    """
+    if any(path.endswith(suffix) for suffix in (".json", ".yaml", ".yml")):
```

Comment:
nit, these can also take a `tuple[str, ...]`
```suggestion
    if path.endswith((".json", ".yaml", ".yml")):
```